Formula/libpulsar.rb

```diff
@@ -0,0 +1,44 @@
+class Libpulsar < Formula
+  desc "Apache Pulsar C++ library"
+  homepage "https://pulsar.apache.org"
+
+  url "https://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename=pulsar/pulsar-2.2.0/apache-pulsar-2.2.0-src.tar.gz"
+  sha256 "a3b1940a803043bb2c365ce9df657d15bf9aacb3c9ff5db86a79dc4374033f08"
+
+  depends_on "boost-python" => :build
+  depends_on "cmake" => :build
+  depends_on "jsoncpp" => :build
+  depends_on "openssl" => :build
+  depends_on "pkg-config" => :build
+  depends_on "protobuf@2.6" => :build
+
+  depends_on "boost"
+
+  def install
+    cd "pulsar-client-cpp" do
+      system "cmake", ".", "-DBUILD_TESTS=OFF", "-DLINK_STATIC=ON",
+          "-DPYTHON_INCLUDE_DIR=#{Formula["python"].include}",
+          "-DBoost_INCLUDE_DIRS=#{Formula["boost"].include}"
+      system "make", "pulsarShared", "pulsarStatic"
```

Comment:
If this is building a shared library, how come the libraries in dependencies (jsonopp, openssl, etc.) are merely build deps, not runtime deps?

Reply:
In general we're statically linking all the dependencies inside libpulsar.so, so that it's independent of versions installed in the system. While that is good in some cases (eg: when building Python binary wheels), I agree that in homebrew context is better to just rely on the runtime dependencies. Switched back to dyn libraries.